File: BackendCitasMedicas/src/main/java/com/example/BackendCitasMedicas/Model/Paciente.java
package com.example.BackendCitasMedicas.Model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.util.List;

@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Paciente {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id_paciente;
    private String nombre;
    private String apellido;
    private LocalDate fecha_nacimiento;
    private String correo;
    private String telefono;
    private String direccion;

    @OneToMany(mappedBy = "paciente")
    @JsonIgnore
    private List<Consulta> consultas;

    @OneToMany(mappedBy = "paciente")
    @JsonIgnore
    private List<Cita> citas;

    @OneToMany(mappedBy = "paciente")
    @JsonIgnore
    private List<Pago> pagos;

}

File: FrontendCitasMedicas/app/src/main/java/com/example/frontcitasmedicas/Model/Paciente.kt
package com.example.frontcitasmedicas.Model

data class Paciente(
    val id_paciente: Long = 0,
    val nombre: String = "",
    val apellido: String = "",
    val numero_identificacion: String = "",
    val fecha_nacimiento: String = "",
    val correo: String = "",
    val telefono: String = "",
    val direccion: String = ""
)
